doc/source/main-documentation/qibolab.rst

```diff
@@ -500,14 +501,48 @@ The shape of the values of an integreted acquisition with 2 sweepers will be:
+The transpiler is responsible for steps 1 and 2, while the compiler for step 3 of the list above. In order to accomplish this, several transpilers are provided, some of which are listed below:
+
+- :class:`qibolab.transpilers.gate_decompositions.NativeGates`: Transpiles single-qubit qibo gates to Z, RZ, GPI2 or U3 and two-qubit gates to CZ and/or iSWAP (depending on platform support).
+- :class:`qibolab.transpilers.star_connectivity.StarConnectivity`: Transforms a circuit to respect a 5-qubit star chip topology, with one middle qubit connected to each of the remaining four qubits.
+- :class:`qibolab.transpilers.routing.ShortestPaths`: Transforms a circuit to respect a general chip topology given as a networkx graph, using a greedy algorithm.
+- :class:`qibolab.transpilers.pipeline.Pipeline`: Applies a list of other transpilers sequentially.
+
+Custom transpilers can be added by inheriting the abstract :class:`qibolab.transpilers.abstract.Transpiler` class.
+
+Once a circuit has been transpiled, it is converted to a :class:`qibolab.pulses.PulseSequence` by the :class:`qibolab.compilers.compiler.Compiler`.
+This is a container of rules which define how each native gate can be translated to pulses.
+A rule is a Python function that accepts a qibo gate and a platform object and returns the :class:`qibolab.pulses.PulseSequence` implementing this gate and a dictionary with potential virtual-Z phases that need to be applied in later pulses.
+Examples of rules can be found on :py:mod:`qibolab.compilers.default`, which defines the default rules used by Qibolab.
+
+.. note::
+   Rules return a :class:`qibolab.pulses.PulseSequence` for each gate, instead of a single pulse, because some gates such as the U3 or two-qubit gates, require more than one pulses to be implemented.
 
 .. _main_doc_native:
 
 Native
 ------
 
+Each quantum platform supports a specific set of native gates, which are the quantum operations that have been calibrated.
+If this set is universal any circuit can be transpiled and compiled to a pulse sequence which is then deployed in the given platform.
+
+:py:mod:`qibolab.native` provides data containers for holding the pulse parameters required for implementing every native gate.
+Every :class:`qibolab.qubits.Qubit` object contains a :class:`qibolab.native.SingleQubitNatives` object which holds the parameters of its native single-qubit gates,
+while each :class:`qibolab.qubits.QubitPair` objects contains a :class:`qibolab.native.TwoQubitNatives` object which holds the parameters of the native two-qubit gates acting on the pair.
+Each native gate is represented by a :class:`qibolab.native.NativePulse` or :class:`qibolab.native.NativeSequence` which contain all the calibrated parameters and can be converted to an actual :class:`qibolab.pulses.PulseSequence` that is then executed in the platform.
+Finally, the :class:`qibolab.native.NativeType` flag is used for communicating the set of available native gates to the transpiler.
```

Comment:
This is technical and looks good, however I am wondering if we should add some extra explanation about the current implementation recipe (pi-pulse, etc...), or at least add some references to the default behaviour of this conversion.